tortoise/verifying_tortoise.go

```diff
@@ -613,34 +621,81 @@ func (t *turtle) processBlock(ctx context.Context, block *types.Block) error {
 		len(block.NeutralDiff) +
 		len(block.NeutralDiff) +
 		len(baseBlockOpinion)
-	opinion := make(map[types.BlockID]vec, lth)
+	opinion := make(Opinion, lth)
 
 	for _, bid := range block.ForDiff {
-		opinion[bid] = support.Multiply(voteWeight)
+		blk, err := t.bdp.GetBlock(bid)
```

Comment:
should be able to look up layerID from `state.BlockLayer` first to avoid accessing disk